openglyph/include/openglyph/io/mega_filesystem.hpp

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include <khepri/io/exceptions.hpp>
+#include <khepri/io/file.hpp>
+#include <khepri/io/stream.hpp>
+
+#include <cassert>
+#include <filesystem>
+#include <map>
+#include <memory>
+
+namespace openglyph::io {
+class SubFile;
+class MegaFile;
+class MegaFileSystem
+{
+public:
+    /**
+     * Constructs a new MegaFileSystem instance.
+     * this class facilitates loading megaFiles
+     * @param data_paths ordered list of paths where to look for assets
+     */
+    MegaFileSystem(std::vector<std::filesystem::path> data_paths);
+
+    std::unique_ptr<SubFile> open_file(std::filesystem::path& path);
```

Comment:
Please add documentation for this method, including a note that the returned Stream may not outlive the MegaFile.